src/ncp/spinel.c

```diff
@@ -525,17 +525,21 @@ static spinel_ssize_t spinel_datatype_vunpack_(bool           in_place,
 
             require_action((spinel_ssize_t)data_len >= (block_len + pui_len), bail, (ret = -1, errno = EOVERFLOW));
 
-            if (in_place)
```

Comment:
In original code, if `block_len` is `0`, it will result in unnecessary call to `memcpy()` for in-place unpacking and update `block_ptr_ptr` to point to invalid buffer for non-in-place unpacking. However I'm OK to keep it as is.